saleor/plugins/vatlayer/plugin.py

```diff
@@ -206,6 +206,65 @@ def calculate_order_line_unit(
             variant.product, order_line.unit_price, country
         )
 
+    def get_checkout_line_tax_rate(
+        self,
+        checkout: "Checkout",
+        checkout_line_info: "CheckoutLineInfo",
+        address: Optional["Address"],
+        discounts: Iterable["DiscountInfo"],
+        previous_value: Decimal,
+    ) -> Decimal:
+        return self._get_tax_rate(checkout_line_info.product, address, previous_value)
+
+    def get_order_line_tax_rate(
+        self,
+        order: "Order",
+        product: "Product",
+        address: Optional["Address"],
+        previous_value: Decimal,
+    ) -> Decimal:
+        return self._get_tax_rate(product, address, previous_value)
+
+    def _get_tax_rate(
+        self, product: "Product", address: Optional["Address"], previous_value: Decimal
+    ):
+        if self._skip_plugin(previous_value):
+            return previous_value
+        country = address.country if address else None
+        taxes, tax_rate = self.__get_tax_data_for_product(product, country)
+        if not taxes or not tax_rate:
+            return previous_value
+        tax = taxes.get(tax_rate) or taxes.get(DEFAULT_TAX_RATE_NAME)
+        # tax value is given in precentage so it need be be converted into decimal value
+        return Decimal(tax["value"] / 100)
+
+    def get_checkout_shipping_tax_rate(
+        self,
+        checkout: "Checkout",
+        lines: Iterable["CheckoutLineInfo"],
+        address: Optional["Address"],
+        discounts: Iterable["DiscountInfo"],
+        previous_value: Decimal,
+    ):
+        return self._get_shipping_tax_rate(address, previous_value)
+
+    def get_order_shipping_tax_rate(self, order: "Order", previous_value: Decimal):
+        address = order.shipping_address or order.billing_address
+        return self._get_shipping_tax_rate(address, previous_value)
+
+    def _get_shipping_tax_rate(
+        self, address: Optional["Address"], previous_value: Decimal
+    ):
+        if self._skip_plugin(previous_value):
+            return previous_value
+        country = address.country if address else None
+        taxes = self._get_taxes_for_country(country)
+        if not taxes:
+            return previous_value
+        tax = taxes.get(DEFAULT_TAX_RATE_NAME)
+        # tax value is given in precentage so it need be be converted into decimal value
+        return Decimal(tax["value"] / 100)
```

Comment:
@IKarbowiak  Not sure of this. What about the case when `tax["value"]` will be 0?

Reply:
It'll work for 0 (as 0/100 is 0) but it should really be `Decimal(tax["value"]) / 100`, otherwise we're creating a `float` and then converting it to a `Decimal`.

Reply:
Good point, I'll change that.